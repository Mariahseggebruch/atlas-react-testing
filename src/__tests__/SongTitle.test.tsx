// SongTitle.test.tsx
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import SongTitle from '../components/SongTitle';

it('renders SongTitle with title and artist', () => {
  const { container } = render(
    <SongTitle title="Song Title" artist="Artist Name" />
  );
  expect(container).toMatchSnapshot();
});

it('renders SongTitle with missing artist', () => {
  const { container } = render(
    <SongTitle title="Song Title" artist="" />
  );
  expect(container).toMatchSnapshot();
});