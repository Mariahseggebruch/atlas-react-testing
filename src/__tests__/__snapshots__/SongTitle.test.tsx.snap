// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders SongTitle with missing artist 1`] = `
<div>
  <div>
    <h2
      class="leading-8 text-2xl font-bold mb-2 text-vista-blue-900"
    >
      Song Title
    </h2>
    <p
      class="mb-4 text-vista-blue-900"
    />
  </div>
</div>
`;

exports[`renders SongTitle with title and artist 1`] = `
<div>
  <div>
    <h2
      class="leading-8 text-2xl font-bold mb-2 text-vista-blue-900"
    >
      Song Title
    </h2>
    <p
      class="mb-4 text-vista-blue-900"
    >
      Artist Name
    </p>
  </div>
</div>
`;
